python/ampli/v1/ampli/ampli.py

```diff
@@ -0,0 +1,889 @@
+        else:
+            env = options.environment or Environment.DEVELOPMENT
+            if options.client:
+                api_key = options.client.api_key or API_KEY[env]
+                self._client = options.client.instance or Amplitude(api_key=api_key)
+                self._client.configuration.api_key = api_key
+                self._client.configuration.server_zone = options.client.zone or SERVER_ZONE
+                self.plan = options.client.plan or AMPLI_OBSERVE_PLAN
+            else:
+                self._client = Amplitude(api_key=API_KEY[env])
+                self.plan = AMPLI_OBSERVE_PLAN
+                self._client.configuration.server_zone = SERVER_ZONE
+        self.configuration = self._client.configuration
+        self.configuration.plan = self.plan
+    
+    def initialized_and_enabled(self) -> bool:
+        if not self._client:
+            logging.getLogger(__name__).error("Ampli is not yet initialized. Called `ampli.load()` on app start.")
+            return False
+        return not self.disabled
+    
+    def track(self, event: BaseEvent, event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        event.load_event_options(event_options)
+        self._client.track(event)
+        
+    def identify(self, user_id: str,
+                 identify_properties: IdentifyProperties,
+                 event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        identify_obj = Identify()
+        if identify_properties:
+            for k, v in identify_properties.properties.items():
+                identify_obj.set(k, v)
+        self._client.identify(identify_obj, event_options)
+
+    def group_identify(self, group_type: str,
+                       group_name: str,
+                       group_properties: GroupProperties,
+                       event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        identify_obj = Identify()
+        if group_properties:
+            for k, v in group_properties.properties.items():
+                identify_obj.set(k, v)
+        self._client.group_identify(group_type, group_name, identify_obj, event_options)
+
+    def set_group(self, user_id: str,
+                  group_type: str,
+                  group_name: Union[str, List[str]],
+                  event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        self._client.set_group(group_type, group_name, event_options)
+
+    def revenue(self, user_id: str,
```

Comment:
revenue? Is that required in Ampli now? cc @justin-fiedler

Reply:
Good call @qingzhuozhen . I think we should remove revenue here and only have load, identify, setGroup, groupIdentify, track, flush. This will reduce the potential conflicts of user defined event method names and "system" methods. They can still send revenue events with track().